### ISSUE TITLE
can't grow ganeti disks on lvm pv's with < 1MB physical extent size

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#403]
Originally reported of Google Code with ID 347.

```
What software version are you running? Please provide the output of "gnt-
cluster --version" and "gnt-cluster version".

What distribution are you using? CentOS 5

<b>What steps will reproduce the problem?</b>
1. Have ganeti use an LVM group with a PE size of 64KB
2. Run gnt-instance grow-disk foobar 0 10g

<b>What is the expected output? What do you see instead?</b>
I would expect the operation to succeed but it fails with:

2013-01-15 16:03:37,934: gnt-instance grow-disk pid=24468 cli:2020 INFO run with arguments '-d asktest1 0 10g'
2013-01-15 16:03:55,554: gnt-instance grow-disk pid=24468 cli:2029 ERROR Error during command processing
Traceback (most recent call last):
  File "/opt/python-2.5/lib/python2.5/site-packages/ganeti/cli.py", line 2025, in GenericMain
    result = func(options, args)
  File "/opt/python-2.5/lib/python2.5/site-packages/ganeti/client/gnt_instance.py", line 651, in GrowDisk
    SubmitOrSend(op, opts)
  File "/opt/python-2.5/lib/python2.5/site-packages/ganeti/cli.py", line 1840, in SubmitOrSend
    return SubmitOpCode(op, cl=cl, feedback_fn=feedback_fn, opts=opts)
  File "/opt/python-2.5/lib/python2.5/site-packages/ganeti/cli.py", line 1817, in SubmitOpCode
    reporter=reporter)
  File "/opt/python-2.5/lib/python2.5/site-packages/ganeti/cli.py", line 1798, in PollJob
    return GenericPollJob(job_id, _LuxiJobPollCb(cl), reporter)
  File "/opt/python-2.5/lib/python2.5/site-packages/ganeti/cli.py", line 1620, in GenericPollJob
    errors.MaybeRaise(msg)
  File "/opt/python-2.5/lib/python2.5/site-packages/ganeti/errors.py", line 486, in MaybeRaise
    raise errcls(*args)
OpExecError: Grow request failed to node foo.bar: Error while executing backend function: integer division or modulo by zero
Failure: command execution error:
Grow request failed to node foo.bar: Error while executing backend function: integer division or modulo by zero


<b>Please provide any additional information below.</b>
The issue appears to be because LogicalVolume.Attach in bdev.py is calling lvs and specifying a unit of megabytes (which returns 0.06 for a 64k PE size) that is then converted to an integer resulting in the value 0. The exception is caused in LogicalVolume.Grow when amount amount is %'d by full_strip_size the value of which is 0. Attached is a patch that uses kb for the units of pe_size and in Grow() converts 'amount' from mb into kb.
```
Originally added on 2013-01-15 21:14:34 +0000 UTC.

Attachments: 
[pe_size_units.patch](https://storage.googleapis.com/google-code-attachments/ganeti/issue-347/comment-0/pe_size_units.patch)

**Comment by Ganeti-Issues-Migrator** [comment]
```
Has anybody had a chance to look at this issue, specifically the attached patch?

Thanks!

-Aaron
```
Originally added on 2013-02-04 15:56:34 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Question: why do you need to support so small PE sizes?
```
Originally added on 2013-02-04 15:59:45 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Good question. It's a long explanation as to why, but our existing ganeti nodes were configured with a 64kb extent size on their volume groups.
```
Originally added on 2013-02-08 22:31:53 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
I hate to be a pest, but has anyone had a chance to look at the patch? Is it likely to be accepted?

Thanks!

-Aaron
```
Originally added on 2013-04-11 21:44:45 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
The patch seems to be OK, but in order for it to be accepted, you should sign the
code.google.com CLA, that you can find here.

https://developers.google.com/open-source/cla/individual

Your name does not appear on the list of signers, so I suppose you've never done it before.
```
Originally added on 2013-04-12 10:27:28 +0000 UTC.

Changed State: Accepted

**Comment by Ganeti-Issues-Migrator** [comment]
```
Sorry for the delay in looking into it, by the way, we had an intense transitioning month. And thanks for looking into signing the CLA so that we can integrate it. I will target the issue at 2.7, our next stable release.
```
Originally added on 2013-04-12 10:41:42 +0000 UTC.

Added to Milestone: Release2.7

**Comment by Ganeti-Issues-Migrator** [comment]
```
Thank you both for the feedback and accepting the patch!

I've signed the CLA.

-Aaron
```
Originally added on 2013-04-12 16:59:47 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2013-04-13 14:39:06 +0000 UTC.

Added Labels: Type-Patch

**Comment by Ganeti-Issues-Migrator** [comment]
```
If you'd be so kind to rebase the patch on top of the latest devel-2.7 branch and resend it patch from git format-patch + git send-email (to ganeti-devel@googlegroups.com) I can apply this with you as "author". Otherwise it's going to be a bit more difficult to apply.
```
Originally added on 2013-04-15 08:44:23 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
I've rebased the patch and sent it to the ganeti-devel list.

-Aaron
```
Originally added on 2013-04-15 16:21:07 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
I just pushed it to the repository. Thanks for your contribution!



commit bc3427b73914a5530d08d55dfd78b595a69c33af
Author: Aaron Knister <aaron.knister@gmail.com>
Date:   Mon Apr 15 12:26:36 2013 -0400

    Use KB as the unit for LVM PE size
    
    LogicalVolume.Attach in bdev.py is calling "lvs" and specifying a unit of
    Megabytes that is then converted to an integer resulting in the value 0 for
    small sizes.
    
    This patch makes Ganeti use KB for the unit instead of MB.
    
    Fixes Issue 347.
    
    Signed-off-by: Aaron Knister <aaron.knister@gmail.com>
    Reviewed-by: Michele Tartara <mtartara@google.com>

```
Originally added on 2013-04-16 09:16:25 +0000 UTC.

Mentioned Issue with GoogleCode ID 347 has been migrated to GitHub with ID #403.
Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
Marking 2.7 bugs fixed, as 2.7.0~rc1 was released, and should be ready to be tested in production environments.
```
Originally added on 2013-05-09 05:42:48 +0000 UTC.

Changed State: Released
